src/CI3Compatible/Database/CI_DB_query_builder.php

```diff
@@ -358,6 +363,27 @@ public function order_by(
         return $this;
     }
 
+    /**
+     * GROUP BY
+     *
+     * @param mixed $by     what field or fields to group_by
+     * @param bool  $escape
+     *
+     * @return CI_DB_query_builder
+     */
+    public function group_by($by, ?bool $escape = null): self
+    {
+        if (is_string($by)) {
+            $by = $escape === true
+                ? explode(',', $by)
+                : [$by];
+        }
```

Comment:
I think it is better to save $by and $escape here, and will pass them to CI4's `groupBy()` method.

At least now the logic `is_bool($escape) OR $escape = $this->_protect_identifiers;` is gone.
https://github.com/bcit-ci/CodeIgniter/blob/45576ef6e62b5ff59da67f1697ee8c57809c7719/system/database/DB_query_builder.php#L1129